src/core/gui/PageView.cpp

```diff
@@ -624,6 +624,20 @@ auto XojPageView::onButtonReleaseEvent(const PositionInputData& pos) -> bool {
 }
 
 auto XojPageView::onKeyPressEvent(GdkEventKey* event) -> bool {
+    if (this->textEditor) {
+        if (this->textEditor->onKeyPressEvent(event)) {
+            return true;
+        }
```

Comment:
Nitpicking, but these can be simplified to
```suggestion
    if (this->textEditor && this->textEditor->onKeyPressEvent(event)) {
        return true;
```

Reply:
The behavior would be a bit different because the use of `else if`. I try to make it close to the original behavior, ie. only one of the textEditor, inputHandler, and verticalSpace gets called.

Reply:
The behavior in the PR is already different from original behavior because it's checking the result of `onKeyPressEvent` instead of returning it directly (the new behavior is better though).

The suggested change is just an equivalent, but simpler version of the new behavior.

Reply:
With my current change, when `textEditor->onKeyPressEvent` returns false `inputHandler->onKeyEvent` does not get called. But with the suggested change, it does get called. (similiar thing with `verticalSpace->onKeyPressEvent`) The first behaviour matches the original was what I meant.

Unless you mean that it is better for the later handlers to be called when the previous ones return false? If that's the case I can update the code.